### PR TITLE
Tiles xyz crs transform

### LINKED ISSUE #52599: If Extent CRS is different than project CRS wrong MBtile area exported
### What is the bug or the crash?

When using the **Generate XYZ tiles (MBtiles)** or **Generate XYZ tiles (Directory)** and the extent is calculated from a layer that is in a different CRS than the project the tiles are created for the wrong area. 

### Steps to reproduce the issue

1. Open up QGIS
2. Add OSM
3. Zoom to an area in the state of Massachusetts USA
4. Add an AOI polygon in CRS in EPSG 3586 
5. Set the project crs to EPSG 26786
6. Run the  Generate XYZ tiles (MBtiles) algorithm with the extent from your AOI polygon layer
7. See that the exported tiles are of the wrong location

### Versions

QGIS version
3.30.0-'s-Hertogenbosch
QGIS code revision
f186b8efe0
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.6.3
PROJ version
9.2.0
EPSG Registry database version
v10.082 (2023-02-06)
GEOS version
3.11.2-CAPI-1.17.2
SQLite version
3.41.1
PDAL version
2.5.2
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows 10 Version 2009

Active Python plugins
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by baswein**
## Description
As described in https://github.com/qgis/QGIS/issues/52599 if the Extent CRS was not the same as the project CRS the Generate XYZ tiles (MBtiles) and Generate XYZ tiles (Directory) algorithms did not output the correct area. This attempts to fix this issue by transforming the extent to the project CRS. 
Please feel free to ignore this if my approach is inappropriate.  
Fixes #52599

### CODE REVIEW

File: python/plugins/processing/algs/qgis/TilesXYZ.py [hunk 1]
```diff
@@ -239,7 +239,14 @@ def generate(self, writer, parameters, context, feedback):
...
         feedback.setProgress(1)
 
-        extent = self.parameterAsExtent(parameters, self.EXTENT, context)
+        project = context.project()
+
+        # Transform extent to match project CRS
+        extent_original = self.parameterAsExtent(parameters, self.EXTENT, context)
+        extent_crs = self.parameterAsExtentCrs(parameters, self.EXTENT, context)
+        self.extentCrsToProjectCrs = QgsCoordinateTransform(extent_crs, project.crs(), context.transformContext())
+        extent = self.extentCrsToProjectCrs.transformBoundingBox(extent_original)
```
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/TilesXYZ.py:248]:
```suggestion
        extent_crs_to_project_crs = QgsCoordinateTransform(extent_crs, project.crs(), context.transformContext())
        extent = extent_crs_to_project_crs.transformBoundingBox(extent_original)
```

**Review by nyalldawson — APPROVED** [review]

**Comment by baswein** [comment]
I see that the auto backport was not successful. Should I try to follow the manual backport instructions?

**Comment by nyalldawson** [comment]
@baswein 
> I see that the auto backport was not successful. Should I try to follow the manual backport instructions?


Yes please! Let's start with a backport for 3.30 and see if the backport bot can then backport that to the ltr release.

**Comment by baswein** [comment]
Ok - just got back from a trip. I'll try that.

**Pull request merged**